syntax = "proto3";


message KERNEL_BINARY
{
    //int32 num_kernels = 1;
    string asic_name = 2;
    message HAWAII
    {
        string driver_name = 4;
        string cl_build_options = 6;
        int32 num_kernels = 8;
        repeated bytes kernel_byte = 9;
        int32 sizeof_kernel_byte = 10;
    }
    repeated HAWAII hawaii = 3;
    message FIJI
    {
        string driver_name = 5;
        string cl_build_options = 7;
    }
    repeated FIJI fiji = 6;
    
}
